apiserver/facades/controller/charmrevisionupdater/charmhub_test.go

```diff
@@ -1,75 +1,56 @@
 // Copyright 2020 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
+// TODO(benhoyt) - add test with some charm resources
+
 package charmrevisionupdater_test
 
 import (
 	"context"
-	"fmt"
 	"time"
 
-	"github.com/juju/charm/v8"
+	"github.com/juju/collections/set"
 	"github.com/juju/errors"
-	"github.com/juju/juju/testcharms"
-	"github.com/juju/names/v4"
+	jujutesting "github.com/juju/testing"
 	jc "github.com/juju/testing/checkers"
 	gc "gopkg.in/check.v1"
 
-	"github.com/juju/juju/apiserver/common"
 	"github.com/juju/juju/apiserver/facades/controller/charmrevisionupdater"
-	"github.com/juju/juju/apiserver/testing"
 	"github.com/juju/juju/charmhub"
 	"github.com/juju/juju/charmhub/transport"
-	jujutesting "github.com/juju/juju/juju/testing"
-	"github.com/juju/juju/state"
 )
 
 type charmhubSuite struct {
-	jujutesting.JujuConnSuite
-
-	updater *charmrevisionupdater.CharmRevisionUpdaterAPI
-	charms  map[string]*state.Charm
+	jujutesting.CleanupSuite
 }
 
 var _ = gc.Suite(&charmhubSuite{})
 
 func (s *charmhubSuite) SetUpSuite(c *gc.C) {
-	s.JujuConnSuite.SetUpSuite(c)
+	s.CleanupSuite.SetUpSuite(c)
 }
 
 func (s *charmhubSuite) SetUpTest(c *gc.C) {
-	s.JujuConnSuite.SetUpTest(c)
-
-	resources := common.NewResources()
-	s.AddCleanup(func(_ *gc.C) { resources.StopAll() })
-	authorizer := testing.FakeAuthorizer{
-		Controller: true,
-		Tag:        names.NewMachineTag("99"),
-	}
-	var err error
-	s.updater, err = charmrevisionupdater.NewCharmRevisionUpdaterAPI(s.State, resources, authorizer)
-	c.Assert(err, jc.ErrorIsNil)
+	s.CleanupSuite.SetUpTest(c)
 
 	// Patch the charmhub initializer function
-	s.PatchValue(&charmrevisionupdater.NewCharmhubClient, func(st *state.State, metadata map[string]string) (charmrevisionupdater.CharmhubRefreshClient, error) {
+	s.PatchValue(&charmrevisionupdater.NewCharmhubClient, func(st charmrevisionupdater.State, metadata map[string]string) (charmrevisionupdater.CharmhubRefreshClient, error) {
```

Comment:
This test file does not need to live in the _test package.  Thus you can avoid using PatchValue.  Or find another way to avoid it.

Reply:
Fixed. Made a new function for testing which takes these dependencies as arguments.